pvlib/irradiance.py

```diff
@@ -3117,3 +3117,72 @@ def complete_irradiance(solar_zenith,
                                      'dhi': dhi,
                                      'dni': dni})
     return component_sum_df
+
+
+def pvl_louche(ghi, solar_zenith, datetime_or_doy):
+    """
+    Determine DNI and GHI from GHI using Louche model.
+
+    Parameters
+    ----------
+    ghi : Series
+        Global horizontal irradiance. [W/m^2]
+
+    solar_zenith : Series
+        True (not refraction-corrected) zenith angles in decimal
+        degrees. Angles must be >=0 and <=90.
+
+    datetime_or_doy : numeric, pandas.DatetimeIndex
+        Day of year or array of days of year e.g.
+        pd.DatetimeIndex.dayofyear, or pd.DatetimeIndex.
+
+    Returns
+    -------
+    data: OrderedDict or DataFrame
+        Contains the following keys/columns:
+
+            * ``dni``: the modeled direct normal irradiance in W/m^2.
+            * ``dhi``: the modeled diffuse horizontal irradiance in
+              W/m^2.
+            * ``kt``: Ratio of global to extraterrestrial irradiance
+              on a horizontal plane.
+    Notes
+    -----
+    `solar_zenith` angles should be between 0 and 90 degrees else NaN will be
+    returned. Day of year (doy) should be between 1 and 366 else NaN will be
+    returned.
+
+    References
+    -------
+    .. [1] Louche A, Notton G, Poggi P, Simmonnot G. Correlations for direct
+       normal and global horizontal irradiation on French Mediterranean site.
+       Solar Energy 1991;46:261-6
+
+    """
+    bool = np.logical_or(solar_zenith > 90, solar_zenith < 0)
+    solar_zenith = np.where(bool, np.NaN, solar_zenith)
+
+    if np.isscalar(datetime_or_doy):
+        bool = (np.any(datetime_or_doy > 366 or datetime_or_doy < 1, axis=0))
+        datetime_or_doy = np.where(bool, np.NaN, datetime_or_doy)
```

Comment:
This code is a bit strange; it uses vectorized functions (`np.any`, `np.where`) but only if the input is scalar?

Anyway we don't do these checks for other decomposition functions so I'm inclined to remove these three lines altogether.